nav2_behavior_tree/include/nav2_behavior_tree/behavior_tree_engine.hpp

```diff
@@ -40,28 +43,29 @@ class BehaviorTreeEngine
     std::function<bool()> cancelRequested,
     std::chrono::milliseconds loopTimeout = std::chrono::milliseconds(10));
 
-  BT::Tree buildTreeFromText(
+  BT::Tree createTreeFromText(
     const std::string & xml_string,
     BT::Blackboard::Ptr blackboard);
 
+  BT::Tree createTreeFromFile(
+    const std::string & file_path,
+    BT::Blackboard::Ptr blackboard);
+
+  void addZMQGrootMonitoring(
+    BT::Tree * tree,
+    uint16_t publisher_port = 1666,
```

Comment:
Remove the defaults here, there's no need for that.

Also remove `zmq` from the name of this function. There's no reason for that.

Reply:
I'll keep the `uint16_t max_msg_per_second = 25` default as it is currently not accessible through ROS2 parameters. I guess the usage pretty low on this. But it is needed to call the BT.CPP function. The bt_navigatior does not even need to list/call it, because when it is at the end of the argument list and lists a default value it can be ignored in the function call.

Reply:
yeah that's fine